drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -521,6 +537,118 @@ func getPortworxServiceSpec(
 	return newService
 }
 
+func (c *portworxBasic) createAndMaintainPxSaTokenSecret(cluster *corev1.StorageCluster, ownerRef *metav1.OwnerReference) error {
+	if err := c.createTokenSecretIfNotExist(cluster, ownerRef); err != nil {
+		return err
+	}
+	if err := c.maintainTokenSecret(cluster, ownerRef); err != nil {
+		return fmt.Errorf("failed to maintain the token secret for px container. %v", err)
+	}
+	return nil
+}
+
+func (c *portworxBasic) createTokenSecretIfNotExist(cluster *corev1.StorageCluster, ownerRef *metav1.OwnerReference) error {
+	secret := &v1.Secret{}
+	err := c.k8sClient.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      pxutil.PortworxServiceAccountTokenSecretName,
+			Namespace: cluster.Namespace,
+		}, secret)
+	if err != nil {
+		if errors.IsNotFound(err) || len(secret.Data) == 0 || len(secret.Data[v1.ServiceAccountTokenKey]) == 0 {
+			token, err := generateToken(cluster)
+			if err != nil {
+				return err
+			}
+			rootCaCrt, _ := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt")
+			curTime := time.Now()
+			tokenRefreshTimeBytes, err := curTime.Add(time.Duration(tokenExpirationSeconds / 2)).MarshalBinary()
+			if err != nil {
+				return fmt.Errorf("error marshalling current time to bytes. %v", err)
+			}
+			secret := &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            pxutil.PortworxServiceAccountTokenSecretName,
+					Namespace:       cluster.Namespace,
+					OwnerReferences: []metav1.OwnerReference{*ownerRef},
+				},
+				Type: v1.SecretTypeOpaque,
+				Data: map[string][]byte{
+					core.ServiceAccountTokenKey:     token,
+					core.ServiceAccountRootCAKey:    rootCaCrt,
+					core.ServiceAccountNamespaceKey: []byte(cluster.Namespace),
+					TokenRefreshTimeKey:             tokenRefreshTimeBytes,
+				},
+			}
+			if err := k8sutil.CreateOrUpdateSecret(c.k8sClient, secret, ownerRef); err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("error getting the token secret for px from k8s. %v", err)
+		}
+	}
+	return nil
+}
+
+func (c *portworxBasic) maintainTokenSecret(cluster *corev1.StorageCluster, ownerRef *metav1.OwnerReference) error {
+	secret := &v1.Secret{}
+	err := c.k8sClient.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      pxutil.PortworxServiceAccountTokenSecretName,
+			Namespace: cluster.Namespace,
+		}, secret)
+	if err != nil {
+		return fmt.Errorf("error getting the secret containing token for px container. %v", err)
```

Comment:
we should probably handle accidental or intentional deletion of the token as well. Maybe "maintain" function should be stateless and do everything (create if not exist, update if stale)? We should probably create an empty secret and then use "refresh" logic to populate it. That way, "refresh" logic gets exercised every time we create the token giving us more coverage from the existing tests. Also, we can avoid some code duplication seen below.